Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,559 @@
+-- ===========================================================================
+-- Base File
+-- ===========================================================================
+include("WorldInput");
+
+-- ===========================================================================
+-- Cached Base Functions
+-- ===========================================================================
+BASE_CQUI_UpdateDragMap            = UpdateDragMap;
+BASE_CQUI_RealizeMovementPath      = RealizeMovementPath;
+BASE_CQUI_OnUnitSelectionChanged   = OnUnitSelectionChanged;
+BASE_CQUI_OnInputActionTriggered   = OnInputActionTriggered;
+BASE_CQUI_DefaultKeyDownHandler    = DefaultKeyDownHandler;
+BASE_CQUI_DefaultKeyUpHandler      = DefaultKeyUpHandler;
+BASE_CQUI_OnDefaultKeyDown         = OnDefaultKeyDown;
+BASE_CQUI_OnDefaultKeyUp           = OnDefaultKeyUp;
+BASE_CQUI_OnPlacementKeyUp         = OnPlacementKeyUp;
+BASE_CQUI_ClearAllCachedInputState = ClearAllCachedInputState;
+BASE_CQUI_LateInitialize           = LateInitialize;
+BASE_CQUI_Initialize               = Initialize;
+
+-- ===========================================================================
+-- CQUI Members
+-- ===========================================================================
+local CQUI_HOTKEYMODE_ENHANCED    = 2;
+local CQUI_HOTKEYMODE_CLASSIC     = 1;
+local CQUI_HOTKEYMODE_STANDARD    = 0;
+
+local m_isInputBlocked  :boolean = false;
+local CQUI_cityview     :boolean = false;
+local CQUI_hotkeyMode   :number  = CQUI_HOTKEYMODE_ENHANCED;
+local CQUI_isShiftDown  :boolean = false;
+local CQUI_isAltDown    :boolean = false;
+local CQUI_isCtrlDown   :boolean = false;
+
+function CQUI_OnSettingsUpdate()
+  CQUI_hotkeyMode = GameConfiguration.GetValue("CQUI_BindingsMode");
+end
+
+LuaEvents.CQUI_SettingsUpdate.Add( CQUI_OnSettingsUpdate );
+LuaEvents.CQUI_SettingsInitialized.Add( CQUI_OnSettingsUpdate );
+
+-- ===========================================================================
+--  VARIABLES
+-- ===========================================================================
+local CQUI_ShowDebugPrint = false;
+
+-- ===========================================================================
+--CQUI setting control support functions
+-- ===========================================================================
+function print_debug(str)
+  if CQUI_ShowDebugPrint then
+    print(str);
+  end
+end
+
+-- ===========================================================================
+-- CQUI Base Extension Functions
+-- Each of these functions call the Base Function found in WorldInput.lua
+-- ===========================================================================
+
+-- ===========================================================================
```

Comment:
Any idea why the original modified this function instead of `StartDragMap`?

Reply:
not sure.  I don't see a reason why it couldn't be in StartDragMap - all the instances where StartDragMap is called, it's preceded by a check whether the mouse/touch is already dragging, which if so, it calls UpdateDragMap.  

That said, this works, and while logically it may make more sense to move it, I'm not sure it's necessary?